homeassistant/components/airvisual/__init__.py

```diff
@@ -205,6 +206,19 @@ async def async_migrate_entry(hass, config_entry):
                     data={CONF_API_KEY: config_entry.data[CONF_API_KEY], **geography},
                 )
             )
+    # 2 -> 3: Explicitly store the integration type in the config entry:
+    if version == 2:
```

Comment:
Instead of doing a migration, let's just check if `CONF_INTEGRATION_TYPE` in entry.data during async_entry_setup. 

That way it will be possible to rollback Home Assistant as it's a soft migration.

Reply:
Let me know if you're okay with 93ded1b; I'll plan on using this mechanism in #34626.

Reply:
Let me know if you're good with ae3b8e4; I'll use this same mechanism in #34626.